### ISSUE TITLE
v3.4.0 error="*rendering.RenderingService run error: Unrecognized remote plugin message:..."

### ISSUE THREAD

**Issue by robonuka** [grafana/grafana-image-renderer#323]
**What happened**: grafana-image-renderer v3.4.0 brings grafana down with error:

` EROR[02-19|12:33:23] Stopped background service               logger=server service=*rendering.RenderingService reason="Unrecognized remote plugin message: \n\nThis usually means that the plugin is either invalid or simply\nneeds to be recompiled to support the latest protocol."
INFO[02-19|12:33:23] Closing tracing                          logger=tracing
WARN[02-19|12:33:23] plugin failed to exit gracefully         logger=plugin.grafana-image-renderer
EROR[02-19|12:33:23] context canceled                         logger=ngalert
EROR[02-19|12:33:23] Server shutdown                          logger=server error="*rendering.RenderingService run error: Unrecognized remote plugin message: \n\nThis usually means that the plugin is either invalid or simply\nneeds to be recompiled to support the latest protocol."
*rendering.RenderingService run error: Unrecognized remote plugin message: 
This usually means that the plugin is either invalid or simply
needs to be recompiled to support the latest protocol. `

**What you expected to happen**: normal work as v3.3.0 does

**How to reproduce it (as minimally and precisely as possible)**: Just install v3.4.0 grafana-image-renderer plugin

**Anything else we need to know?**:

**Environment**: self-made docker image
- Grafana Image Renderer version: v3.4.0
- Grafana version: grafana/grafana:main or grafana/grafana:8.4.1
- Installed plugin or remote renderer service: plugin installed with
`grafana-cli --pluginsDir "$GF_PATHS_PLUGINS" --pluginUrl https://github.com/grafana/grafana-image-renderer/releases/latest/download/plugin-linux-x64-glibc-no-chromium.zip glibc-no-chromium.zip plugins install grafana-image-renderer`
- OS Grafana Image Renderer is installed on: Alpine (original OS in the grafana/grafana:XXX docker images)
- User OS & Browser: any (grafana container goes down while starting)
- Others: if I downgrade grafana-image-renderer plugin to v3.3.0 - everything is fine. Grafana works as usually.

**Comment by ArturWierzbicki** [comment]
Hey @robonuka, thank you for reporting this issue! We have added dependency on `sharp` image processing library in 3.4.0 and that's the root cause of this issue. 

Under the hood, sharp uses [libvips](https://github.com/libvips/libvipshttps://github.com/libvips/libvips) which comes precompiled for the most popular platforms. Unfortunately, the `linux-x64-glibc` version of Image Renderer is incompatible with docker alpine, because docker alpine uses `musl` rather than `glibc`. 

I have added _alpine-compatible_ linux-musl build in [this](https://github.com/grafana/grafana-image-renderer/pull/324) PR and updated the alpine-based Grafana + Image Renderer dockerfile in [here](https://github.com/grafana/grafana/pull/45644).

`3.4.0` image render should work just fine with the ubuntu image -https://github.com/grafana/grafana/blob/main/packaging/docker/custom/ubuntu.Dockerfile)

**Comment by robonuka** [comment]
Hello Artur!
I rebuilt my docker image from the original grafana "-ubuntu" image as source.
v3.4.0 renderer-plugin doesn't bring grafana down, but in the grafana container log I see this error message:
`logger=plugin.grafana-image-renderer t=2022-02-21T07:18:22.76+0300 lvl=eror msg="Failed to get browser version" err="unsupported value type"`
Tell me please how to fix this error?

**Comment by ArturWierzbicki** [comment]
Hey @robonuka, the documentation (https://grafana.com/grafana/plugins/grafana-image-renderer/) mentions that the plugin is not compatible with the official docker image:

> This plugin is not compatible with the current Grafana Docker image and requires additional system-level dependencies. We recommend setting up another Docker container for rendering and using remote rendering instead. For instruction, refer to [Run in Docker](https://grafana.com/grafana/plugins/grafana-image-renderer/#run-in-docker).

> If you still want to install the plugin with the Grafana Docker image, refer to the instructions on building a custom Grafana image in [Grafana Docker documentation](https://grafana.com/docs/installation/docker/#custom-image-with-grafana-image-renderer-plugin-pre-installed).


I followed linked instructions from https://grafana.com/docs/grafana/latest/installation/docker/#build-with-grafana-image-renderer-plugin-pre-installed to make it work. The dockerfile: https://github.com/grafana/grafana/blob/main/packaging/docker/custom/ubuntu.Dockerfile

**Comment by robonuka** [comment]
I found the reason of this problem. I forgot to add variable `GF_PLUGIN_RENDERING_CHROME_BIN=/usr/bin/google-chrome` to the environments of grafana docker container because by default renderer plugin looking for chromium browser as `/usr/bin/chromium-browser`
Now error `msg="Failed to get browser version"` has gone. Everything working fine! Thank you!

**Comment by ZdenekM** [comment]
Strange... I just updated to 3.4.0 (Grafana 8.4.1 installed using apt) and there is the same error:

```
úno 22 14:08:13 AtomNAS grafana-server[12096]: logger=settings t=2022-02-22T14:08:13.63+0100 lvl=info msg="Starting Grafana" version=8.4.1 commit=53f5c6a44c bran
úno 22 14:08:13 AtomNAS grafana-server[12096]: logger=settings t=2022-02-22T14:08:13.63+0100 lvl=warn msg="falling back to legacy setting of 'min_interval_second
úno 22 14:08:13 AtomNAS grafana-server[12096]: logger=settings t=2022-02-22T14:08:13.63+0100 lvl=info msg="Config loaded from" file=/usr/share/grafana/conf/defau
úno 22 14:08:13 AtomNAS grafana-server[12096]: logger=settings t=2022-02-22T14:08:13.63+0100 lvl=info msg="Config loaded from" file=/etc/grafana/grafana.ini
úno 22 14:08:13 AtomNAS grafana-server[12096]: logger=settings t=2022-02-22T14:08:13.63+0100 lvl=info msg="Config overridden from command line" arg="default.path
úno 22 14:08:13 AtomNAS grafana-server[12096]: logger=settings t=2022-02-22T14:08:13.63+0100 lvl=info msg="Config overridden from command line" arg="default.path
úno 22 14:08:13 AtomNAS grafana-server[12096]: logger=settings t=2022-02-22T14:08:13.63+0100 lvl=info msg="Config overridden from command line" arg="default.path
úno 22 14:08:13 AtomNAS grafana-server[12096]: logger=settings t=2022-02-22T14:08:13.63+0100 lvl=info msg="Config overridden from command line" arg="default.path
úno 22 14:08:13 AtomNAS grafana-server[12096]: logger=settings t=2022-02-22T14:08:13.63+0100 lvl=info msg="Path Home" path=/usr/share/grafana
úno 22 14:08:13 AtomNAS grafana-server[12096]: logger=settings t=2022-02-22T14:08:13.63+0100 lvl=info msg="Path Data" path=/var/lib/grafana
úno 22 14:08:13 AtomNAS grafana-server[12096]: logger=settings t=2022-02-22T14:08:13.63+0100 lvl=info msg="Path Logs" path=/var/log/grafana
úno 22 14:08:13 AtomNAS grafana-server[12096]: logger=settings t=2022-02-22T14:08:13.63+0100 lvl=info msg="Path Plugins" path=/var/lib/grafana/plugins
úno 22 14:08:13 AtomNAS grafana-server[12096]: logger=settings t=2022-02-22T14:08:13.63+0100 lvl=info msg="Path Provisioning" path=/etc/grafana/provisioning
úno 22 14:08:13 AtomNAS grafana-server[12096]: logger=settings t=2022-02-22T14:08:13.63+0100 lvl=info msg="App mode production"
úno 22 14:08:13 AtomNAS grafana-server[12096]: logger=sqlstore t=2022-02-22T14:08:13.64+0100 lvl=info msg="Connecting to DB" dbtype=sqlite3
úno 22 14:08:13 AtomNAS grafana-server[12096]: logger=migrator t=2022-02-22T14:08:13.86+0100 lvl=info msg="Starting DB migrations"
úno 22 14:08:13 AtomNAS grafana-server[12096]: logger=migrator t=2022-02-22T14:08:13.96+0100 lvl=info msg="migrations completed" performed=0 skipped=381 duration
úno 22 14:08:14 AtomNAS grafana-server[12096]: logger=plugin.manager t=2022-02-22T14:08:14.37+0100 lvl=info msg="Plugin registered" pluginId=input
úno 22 14:08:26 AtomNAS grafana-server[12096]: logger=plugin.manager t=2022-02-22T14:08:26.87+0100 lvl=info msg="Plugin registered" pluginId=grafana-image-render
úno 22 14:08:26 AtomNAS grafana-server[12096]: logger=query_data t=2022-02-22T14:08:26.87+0100 lvl=info msg="Query Service initialization"
úno 22 14:08:26 AtomNAS grafana-server[12096]: logger=live.push_http t=2022-02-22T14:08:26.89+0100 lvl=info msg="Live Push Gateway initialization"
úno 22 14:08:28 AtomNAS grafana-server[12096]: logger=server t=2022-02-22T14:08:28.25+0100 lvl=info msg="Writing PID file" path=/run/grafana/grafana-server.pid p
úno 22 14:08:28 AtomNAS grafana-server[12096]: logger=ngalert t=2022-02-22T14:08:28.26+0100 lvl=info msg="warming cache for startup"
úno 22 14:08:28 AtomNAS grafana-server[12096]: logger=ngalert.multiorg.alertmanager t=2022-02-22T14:08:28.34+0100 lvl=info msg="starting MultiOrg Alertmanager"
úno 22 14:08:28 AtomNAS grafana-server[12096]: logger=http.server t=2022-02-22T14:08:28.36+0100 lvl=info msg="HTTP Server Listen" address=[::]:3000 protocol=http
úno 22 14:08:31 AtomNAS grafana-server[12096]: logger=server t=2022-02-22T14:08:31.5+0100 lvl=eror msg="Stopped background service" service=*rendering.RenderingS
úno 22 14:08:31 AtomNAS grafana-server[12096]: logger=tracing t=2022-02-22T14:08:31.5+0100 lvl=info msg="Closing tracing"
úno 22 14:08:31 AtomNAS grafana-server[12096]: logger=ngalert t=2022-02-22T14:08:31.5+0100 lvl=eror msg="unable to fetch orgIds" msg="context canceled"
úno 22 14:08:31 AtomNAS grafana-server[12096]: logger=plugin.grafana-image-renderer t=2022-02-22T14:08:31.5+0100 lvl=warn msg="plugin failed to exit gracefully"
úno 22 14:08:31 AtomNAS grafana-server[12096]: logger=server t=2022-02-22T14:08:31.5+0100 lvl=eror msg="Server shutdown" error="*rendering.RenderingService run e
úno 22 14:08:31 AtomNAS grafana-server[12096]: *rendering.RenderingService run error: Unrecognized remote plugin message:
úno 22 14:08:31 AtomNAS grafana-server[12096]: This usually means that the plugin is either invalid or simply
úno 22 14:08:31 AtomNAS grafana-server[12096]: needs to be recompiled to support the latest protocol.
úno 22 14:08:31 AtomNAS systemd[1]: grafana-server.service: Main process exited, code=exited, status=1/FAILURE
úno 22 14:08:31 AtomNAS systemd[1]: grafana-server.service: Failed with result 'exit-code'.
úno 22 14:08:31 AtomNAS systemd[1]: grafana-server.service: Service hold-off time over, scheduling restart.
úno 22 14:08:31 AtomNAS systemd[1]: grafana-server.service: Scheduled restart job, restart counter is at 1.
úno 22 14:08:31 AtomNAS systemd[1]: Stopped Grafana instance.
```

**Comment by ArturWierzbicki** [comment]
hey @ZdenekM, thanks for reporting the issue. 

What is the name&version of your OS? Could you please run `/var/lib/grafana/plugins/grafana-image-renderer/plugin_start_linux_amd64` and paste the output?

**Comment by ZdenekM** [comment]
@ArturWierzbicki Ubuntu 18.04.6 (kernel 4.15.0-166-generic, x86_64). The output is `Illegal instruction (SIGILL)`. From gdb, I got:

```
Thread 1 "plugin_start_li" received signal SIGILL, Illegal instruction.
0x00007fffdf139dfb in ?? () from /var/lib/grafana/plugins/grafana-image-renderer/node_modules/sharp/build/Release/../.././vendor/8.11.3/linux-x64/lib/libvips-cpp.so.42
```

**Comment by ArturWierzbicki** [comment]
@ZdenekM We have just released 3.4.1 which should fix that issue!

**Comment by ZdenekM** [comment]
@ArturWierzbicki Grafana now starts, thanks!
